core/src/main/java/io/grpc/transport/AbstractClientStream.java

```diff
@@ -125,8 +130,8 @@ protected void inboundHeadersReceived(Metadata.Headers headers) {
    *
    * @param frame the received data frame. Its ownership is transferred to this method.
    */
-  protected void inboundDataReceived(ReadableBuffer frame) {
-    Preconditions.checkNotNull(frame, "frame");
```

Comment:
So it sounds like there might be two things here.
1. There could be something that includes the version number in the stack trace  (something that catches it at a very high level, and tacks it on).  If knowing the version number is important, then a stray string in the message doesn't seem like a good way of propagating it.

2.  If the goal is to make it easy for users to see where they messed up, then a sentence fragment of what was expected would probably be more appropriate.  At least for internal APIs, the user may have missed providing some higher level parameter, which only caused indirectly (frame in this case) to be not set.  A full error message would be a better solution.

As for checkNotNull, A shorter name also can mean the difference between having to wrap a line and not.